### ISSUE TITLE
unittest.makeSuite is deprecated in Python 3.11 and will be removed in Python 3.13

### ISSUE THREAD

**Issue by tirkarthi** [zopefoundation/zope.security#83]
## BUG/PROBLEM REPORT (OR OTHER COMMON ISSUE)

Ref : python/cpython#28382

```python
src/zope/security/tests/test_zcml.py
196:        unittest.makeSuite(PermissionTests),
197:        unittest.makeSuite(Test_securityPolicy),
198:        unittest.makeSuite(Test_permission),
199:        unittest.makeSuite(Test_redefinePermission),

src/zope/security/tests/test_protectclass.py
144:        unittest.makeSuite(Test_protectName),
145:        unittest.makeSuite(Test_protectSetAttribute),
146:        unittest.makeSuite(Test_protectLikeUnto),

src/zope/security/tests/test_simpleinteraction.py
79:        unittest.makeSuite(TestInteraction),
```

**Comment by icemac** [comment]
Thank you for creating this issue. We'll have to fix it in quite all zopefoundation projects.

**Comment by jugmac00** [comment]
> ```python
> makeSuite
> ```

$ all-repos-grep --repos-with-matches makeSuite
repos/zopefoundation/AccessControl
repos/zopefoundation/DateTime
repos/zopefoundation/DocumentTemplate
repos/zopefoundation/Products.CMFCore
repos/zopefoundation/Products.CMFDefault
repos/zopefoundation/Products.CMFUid
repos/zopefoundation/Products.DCWorkflow
repos/zopefoundation/Products.GenericSetup
repos/zopefoundation/Products.MIMETools
repos/zopefoundation/Products.MailHost
repos/zopefoundation/Products.PluggableAuthService
repos/zopefoundation/Products.PluginRegistry
repos/zopefoundation/Products.Sessions
repos/zopefoundation/Products.ZMySQLDA
repos/zopefoundation/Products.ZSQLMethods
repos/zopefoundation/Products.ZopeVersionControl
repos/zopefoundation/ZEO
repos/zopefoundation/ZODB
repos/zopefoundation/ZServer
repos/zopefoundation/Zope
repos/zopefoundation/cipher.encryptingstorage
repos/zopefoundation/roman
repos/zopefoundation/tempstorage
repos/zopefoundation/transaction
repos/zopefoundation/z3c.authenticator
repos/zopefoundation/z3c.form
repos/zopefoundation/z3c.jbot
repos/zopefoundation/z3c.menu.ready2go
repos/zopefoundation/z3c.objpath
repos/zopefoundation/z3c.recipe.compattest
repos/zopefoundation/z3c.relationfield
repos/zopefoundation/z3c.rml
repos/zopefoundation/z3c.schema
repos/zopefoundation/z3c.sqlalchemy
repos/zopefoundation/z3c.testing
repos/zopefoundation/z3c.wizard
repos/zopefoundation/zc.intid
repos/zopefoundation/zc.recipe.testrunner
repos/zopefoundation/zc.zlibstorage
repos/zopefoundation/zc.zrs
repos/zopefoundation/zdaemon
repos/zopefoundation/zodbpickle
repos/zopefoundation/zope.app.appsetup
repos/zopefoundation/zope.app.authentication
repos/zopefoundation/zope.app.container
repos/zopefoundation/zope.app.form
repos/zopefoundation/zope.app.http
repos/zopefoundation/zope.app.pagetemplate
repos/zopefoundation/zope.app.publication
repos/zopefoundation/zope.app.testing
repos/zopefoundation/zope.authentication
repos/zopefoundation/zope.browserpage
repos/zopefoundation/zope.browserresource
repos/zopefoundation/zope.component
repos/zopefoundation/zope.configuration
repos/zopefoundation/zope.container
repos/zopefoundation/zope.copypastemove
repos/zopefoundation/zope.datetime
repos/zopefoundation/zope.dottedname
repos/zopefoundation/zope.dublincore
repos/zopefoundation/zope.formlib
repos/zopefoundation/zope.i18n
repos/zopefoundation/zope.index
repos/zopefoundation/zope.intid
repos/zopefoundation/zope.location
repos/zopefoundation/zope.proxy
repos/zopefoundation/zope.publisher
repos/zopefoundation/zope.renderer
repos/zopefoundation/zope.security
repos/zopefoundation/zope.site
repos/zopefoundation/zope.sqlalchemy
repos/zopefoundation/zope.tal
repos/zopefoundation/zope.testing
repos/zopefoundation/zope.testrunner
repos/zopefoundation/zope.traversing
repos/zopefoundation/zope.untrustedpython

**Comment by sallner** [comment]
It looks like `unittest.TestLoader.loadTestsFromTestCase` is already available under python 2.7, so we can simply exchange that.

**Comment by d-maurer** [comment]
Steffen Allner wrote at 2022-4-13 01:10 -0700:
>It looks like `unittest.TestLoader.loadTestsFromTestCase` is already available under python 2.7, so we can simply exchange that.

`makeSuite` is used by many `zopefoundation` packages (even though
it is often not necessary to control the test selection).

An alternative to updating all those packages could be to
change `zope.testrunner` to monkey-readd `makeSuite` to `unittest`.

**Comment by sallner** [comment]
> `makeSuite` is used by many `zopefoundation` packages (even though it is often not necessary to control the test selection). An alternative to updating all those packages could be to change `zope.testrunner` to monkey-readd `makeSuite` to `unittest`.

That might be true, but in general I would prefer to use the new world if possible. In case we would use `python-modernize` or `pyupgrade` to remove Python 2 compatibility code, it would be nice to automate this step also.
